docs/property-grid.md

```diff
@@ -0,0 +1,418 @@
+
+Numeric values that specify the minimum and maximum numbers users can enter into the editor.
+
+```js
+Survey.Serializer.addProperty("question",
+  { name: "my-numeric-property", type: "number", minValue: 0, maxValue: 100 }
+);
+```
+
+#### `dependsOn`
+
+An array of property names upon which the current property depends. When one of the listed properties changes, the dependent property reevaluates the [`visibleIf`](#visibleif) and [`choices`](#choices) functions. This allows you to control the property visibility and fill choices conditionally.
+
+The following code declares two custom properties. `dependent-property` fills `choices` depending on the `my-custom-property` value:
+
+```js
+Survey.Serializer.addProperty("question", {
+  name: "my-custom-property",
+  choices: ["Option 1", "Option 2", "Option 3"],
+});
+
+Survey.Serializer.addProperty("question", {
+  name: "dependent-property",
+  dependsOn: [ "my-custom-property" ],
+  choices: function (obj) {
+    const choices = [];
+    const targetPropertyValue = !!obj ? obj["my-custom-property"] : null;
+    // If `targetPropertyValue` is empty, return an empty array
+    if (!targetPropertyValue) return choices;
+    // Make the dependent property nullable
+    choices.push({ value: null });
+    // Populate `choices`
+    choices.push(targetPropertyValue + ": Suboption 1");
+    choices.push(targetPropertyValue + ": Suboption 2");
+    choices.push(targetPropertyValue + ": Suboption 3");
+    return choices;
+  }
+});
+```
+
+The following example shows how to load `choices` for the `country` property from a web service. They are reloaded each time a user changes the `region` value:
+
+```js
+Survey.Serializer.addProperty("survey", {
+  name: "region",
+  category: "Region",
+  categoryIndex: 1,
+  choices: ["Africa", "Americas", "Asia", "Europe", "Oceania"],
+});
+
+Survey.Serializer.addProperty("survey", {
+  name: "country",
+  category: "Region",
+  dependsOn: [ "region" ],
+  choices: function (obj, choicesCallback) {
+    const xhr = new XMLHttpRequest();
+    const url =
+      !!obj && !!obj.region
+        ? "https://surveyjs.io/api/CountriesExample?region=" + obj.region
+        : "https://surveyjs.io/api/CountriesExample";
+    xhr.open("GET", url);
+    xhr.setRequestHeader("Content-Type","application/x-www-form-urlencoded");
+    xhr.onload = function () {
+      if (xhr.status === 200) {
+        const response = JSON.parse(xhr.response);
+        const result = [];
+        // Make the property nullable
+        res.push({ value: null });
+        response.forEach(item => {
+          result.push({ value: item.cioc, text: item.name });
+        });
+        choicesCallback(result);
+      }
+    };
+    xhr.send();
+  }
+});
+```
+
+#### `onGetValue`
+
+A function that you can use to adjust or exclude the property value from the survey JSON definition.
+
+```js
+// get title property returns a title with question number and so on "5) My super title.",
```

Comment:
Do you mean the following?

// The get title property returns a title with its question number (for example, "5) My super title."),